src/RoomsSDKAdapter.js

```diff
@@ -126,12 +135,56 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
       );
 
       // Convert to a multicast observable
-      this.getRoomObservables[ID] = getRoom$.pipe(
-        publishReplay(1),
-        refCount(),
-      );
+      this.getRoomObservables[ID] = getRoom$.pipe(publishReplay(1), refCount());
     }
 
     return this.getRoomObservables[ID];
   }
+
+  /**
+   * Returns an observable that emits current and future activities from the specified room.
+   *
+   * @param {string} id ID of room to get
+   * @returns {Observable.<Activity>} Observable stream that emits current and future activities from the specified room
+   */
+  getRoomActivities(id) {
+    logger.debug('ROOM', id, 'getRoomActivities()', ['called with', {id}]);
+
+    // if (!(ID in this.getRoomObservables)) {
+    //   return throwError(new Error(`Room ${ID} does not exist`));
+    // }
+
+    const activity$ = new BehaviorSubject(null);
+
+    this.datasource.internal.mercury.on(
+      'event:conversation.activity',
+      (activity) => {
+        const {id: hydraId} = deconstructHydraId(id);
+
+        if (activity.target && activity.target.id === hydraId) {
+          const activityInterface = {
+            id: activity.id,
+            roomId: activity.target.id,
+            verb: activity.verb,
+            content: activity.object,
+            contentType: activity.object.objectType,
+            personId: activity.actor.id,
+            displayAuthor: !!activity.object.displayName,
+            created: activity.published,
+          };
+
+          activity$.next(activityInterface);
+        }
+      },
+    );
+
+    return activity$.pipe(
+      // skip(1), // causes test case to fail, use filter instead to skip first value
+      filter((activity) => activity !== null),
```

Comment:
Not sure i understand why you're filtering

Reply:
Filtering is done so it doesnt send an empty object when you first subscribe

Reply:
it shouldn't send `null` ever, it should always send an activity object